File: Tungsten/Models/Course.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tungsten.Models
{
    public class Course : Entity<string>, DescribedSection
    {
        public Course()
        {
            Id = Guid.NewGuid().ToString();
        }

        [Key]
        public string Id { get; set; }
        [Display(Name = "Namn")]
        public string Name { get; set; }
        [Display(Name = "Beskrivning")]
        public string Description { get; set; }

        [Required, ForeignKey("Group")]
        public string GroupId { get; set; }
        public Group Group { get; set; }

        public virtual ICollection<Segment> Segments { get; set; }
        public virtual ICollection<ApplicationUser> Teachers { get; set; }
        public virtual ICollection<Lesson> Lessons { get; set; }
    }
}

File: Tungsten/LMS/classes/Course.ts
import { ICourse } from '../interfaces/Course';

export class Course implements ICourse {
    Id: string;
    Name: string;
    Description: string;
    Subject: string;
    Level: string;
    GroupId: string;

    constructor(name: string, description: string, subject: string, level: string, groupid: string) {
        this.Name = name;
        this.Description = description;
        this.Subject = subject;
        this.Level = level;
        this.GroupId = groupid;
    }
}
